### ISSUE TITLE
No UMAP separation and loss of structure when training at higher resolution?

### ISSUE THREAD

**Issue by Joshdudley** [ml-struct-bio/cryodrgn#484]
I’m running v3.4.4 and training on 871k particles that reconstruct in cryoSPARC to ~4 angstroms. In cryoDRGN I am getting no separation in the UMAP analysis. Training downsampled particles to 64-pixels gives kmeans30 volumes with each resembling the particle. Increasing to 128-pixels results in many noise-only volumes (with concentric density in the center of the box) and at 256-pixels all the volumes are noise. There is smooth decay in the learning curves for all runs so I'm wondering if there may be something other than the learning rate I may be overlooking.

I’ve also tried zdim 8,16,32,64 and found no significant improvements. Any advice of what I might try next to achieve separation of states? 

Please see the attached config file along with the clusters at increasing pixel-size and png for UMAP learning. Here’s an example of the code I have been running:

    cryodrgn train_vae \
      "${particle_stack}" \
      --poses "${poses_file}" \
      --ctf "${ctf_file}" \
      --zdim "${zdim}" \
      -n 50 \
      --multigpu \
      --shuffle-seed 1 \
      -o runs/main_cs_${boxsize}

**Comment by Joshdudley** [comment]
[config.yaml](https://github.com/user-attachments/files/30793229/config.yaml)

<img width="913" height="659" alt="Image" src="https://github.com/user-attachments/assets/70a35448-a7b9-4e9a-a0d1-7d6db0ea0614" />

<img width="913" height="659" alt="Image" src="https://github.com/user-attachments/assets/b91e3f3b-5798-4d7c-aa30-99f59b1de51c" />

<img width="913" height="659" alt="Image" src="https://github.com/user-attachments/assets/99f61863-e836-42a4-a1a2-e6e2196e7184" />

<img width="600" height="600" alt="Image" src="https://github.com/user-attachments/assets/60b234a9-db65-4880-9b6f-6745a0f81330" />

<img width="640" height="480" alt="Image" src="https://github.com/user-attachments/assets/72e2f9dc-e7c4-4f91-885b-9a87fadf767b" />

**Comment by michal-g** [comment]
Hi Josh, sorry to hear you are having trouble! In this situation my first guess would be either overfitting or underfitting, and to try a smaller model on D=128 (`--enc-dim` or `--dec-dim` or both to 256 instead of the default 1024) and possibly *fewer* latent dimensions (4 instead of 8). I assume that in your successful reconstruction you are seeing compositional and not conformational heterogeneity — what are the relative sizes of the different clusters? Is it junk vs. not junk or a number of states?

In your message you mention looking at `kmeans30` volumes, but the training command you shared with us uses 50 epochs — are these results and images from the 50th epoch of this command's run? If you are already seeing the same results after e.g. 5 or 10 epochs that would further support the presence of overfitting!

If switching to a smaller model doesn't work, and you don't see any further evidence of overfitting, I would try a larger model next (`--enc-dim`/`--dec-dim` to 2048, `enc-layers`/`dec-layers` to 5 or 10).

**Comment by Joshdudley** [comment]
Hi Michal! Thanks for getting back to me. I appreciate it.

This sample is for an apo protein so I'm only expecting to see conformational heterogeneity. The clusters have been pretty uniformly distributed (~25 to 30k each). None of the classes at D=64 are obvious junk. It's a little harder to tell at D=128 with the odd density in the middle of the box but also no blatant junk clusters. At D=256 it's all noise. The results and images are from the kmeans30 of the 40th epoch. To my eye there's no major difference between that and the final epoch. And to make sure I understand, when you mention seeing results after 5 or 10 epochs, if it's overfitting would you expect to see protein-like volumes that don't differ much from the later epochs? 

I'll start a smaller model run at D=128 with --enc-dim and --dec-dim set to 256 are start checking the earlier epochs.

**Comment by michal-g** [comment]
Have you tried looking at the PC1/PC2 volume trajectories? In the case of conformational heterogeneity you may not always see a cleanly-separated latent space, or even one that isn't a big blob, but moving across the latent space can still reveal that different co-ordinates map to different conformations. You can also look at the `kmeans` volumes for another type of cross-section of volumes across the latent space — is the volume you are showing the one of the `kmeans` volumes?

I misread your original comment about `kmeans30`, which would correspond to using `--ksample 30` with `analyze` and not `analyze` on the 30th epoch — but yes, that's correct, if you are already seeing the same results very early in training it's a good sign the model is overpowered! On the other hand, results getting worse as input resolution increases is suggestive of the model being underpowered, and if your compute allows I might try a bigger model on D=256 in parallel with your current run.

**Comment by Joshdudley** [comment]
I haven't looked at those volumes before. The first 2 components have been surprisingly gaussian across all runs. I'll take a look and would you suggest running the analysis with additional components? And if the latent space doesn't separate, would you then try to sample the blob uniformly? The volumes I showed are the 30 kmeans volumes overlaid- apologies if that was misleading- I was only trying to convey that structure has been lost when increasing D. I'm on a computer cluster but am throttled a bit so I'll try the 256 next. Thanks for the chat so far!

**Comment by ryanfeathers** [comment]
Hi Josh,

Thanks for sharing all of this info! I second Michal's suggestion of trying smaller --enc-dim and --dec-dim as well as looking at earlier epochs and the PC volumes. 

I also wanted to ask, what is your pixel size for the 64 pixel box? I usually try to downsample to around 3Å/pix. This often results in a 128 pixel box except when samples are very large or very small.

 I'm also curious, were these particles heavily classified prior to cryoDRGN analysis? If you're achieving a relatively uniform 4Å map in cryosparc you may have already filtered out the conformations you are interested in. I recommend starting with a particle stack that has had minimal 3D classification but averages to a reasonable (<8Å) consensus volume.
